server/activityItem/model.ts

```diff
@@ -9,34 +9,44 @@ import {
 	Index,
 } from 'sequelize-typescript';
 import type { InferAttributes, InferCreationAttributes, CreationOptional } from 'sequelize';
-import { ActivityItemKind, ActivityItemPayload } from 'types';
+import { InsertableActivityItem, RecursiveAttributes } from 'types';
 
 @Table
-export class ActivityItem extends Model<
-	InferAttributes<ActivityItem>,
-	InferCreationAttributes<ActivityItem>
+// export class ActivityItem<Kind extends ActivityItemKind = ActivityItemKind> extends Model<
+// 	InferAttributes<ActivityItem<Kind>>,
+// 	InferCreationAttributes<ActivityItem<Kind>>
+// > {
+export class ActivityItem<T extends InsertableActivityItem = InsertableActivityItem> extends Model<
```

Comment:
This was necessary to turn activityItems  into a discriminated union of `{kind: 'pub-created', payload: { pub: ... } } | { kind: 'collection-created', payload: { collection: ... } }` instead of 
`{ kind: 'collection-created' | 'pub-created' | ..., payload: { pub: ... } | { collection: ... } | ... }`